typing/includecore.ml

```diff
@@ -207,11 +227,16 @@ let report_constructor_mismatch first second decl ppf err =
 let report_variant_mismatch first second decl ppf err =
   let pr fmt = Format.fprintf ppf fmt in
   match (err : variant_mismatch) with
-  | Constructor_mismatch err ->
-    report_constructor_mismatch first second decl ppf err
+  | Constructor_mismatch (c1, c2, err) ->
+      pr
+        "@[<hv>Constructors do not match:@;<1 2>%a@ is not compatible with:\
+         @;<1 2>%a@ %a"
+        Printtyp.constructor c1
+        Printtyp.constructor c2
+        (report_constructor_mismatch first second decl) err
   | Constructor_names (n, name1, name2) ->
-      pr "Constructors number %i have different names, %s and %s."
-        n (Ident.name name1) (Ident.name name2)
+      pr "%i%s constructors have different names, %s and %s."
```

Comment:
I would really prefer if we avoided starting sentences with a number.

Reply:
Rolled back changes for this message and changes connected to suffix function as well